test/e2e/lb_subnets_test.go

```diff
@@ -0,0 +1,419 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"testing"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/api/features"
+	machinev1 "github.com/openshift/api/machine/v1beta1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/cluster-ingress-operator/pkg/operator/controller"
+	"github.com/openshift/cluster-ingress-operator/pkg/operator/controller/ingress"
+
+	v12 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	awsLBSubnetAnnotation = "service.beta.kubernetes.io/aws-load-balancer-subnets"
+)
+
+// TestAWSLBSubnets creates an IngressController with both the public and private cluster subnets for its LB-type
```

Comment:
Have we added a test to check what happens when an invalid or non-exisiting subnet is provided in the IC CR ?

Reply:
We have a test which provides a valid private subnet to a public IngressController, which breaks the IngressController, but as far as non-existing subnets, we don't.

It's a good idea and pretty easy to do, I'll add one.

Reply:
in my orig PR 
invalid EIP : https://github.com/openshift/cluster-ingress-operator/pull/1034/commits/86f94a54e77f46496565b6ec0b38526ab08731e0#diff-160d46f77f47e8e0d35737b019c27a6e4a6f7e7eae4f6e85c0dcdb43b20a2379R445
non-existent EIP: https://github.com/openshift/cluster-ingress-operator/pull/1034/commits/86f94a54e77f46496565b6ec0b38526ab08731e0#diff-160d46f77f47e8e0d35737b019c27a6e4a6f7e7eae4f6e85c0dcdb43b20a2379R558